File: cases/taint/go/nested-access-path-negative/nested_access_path_negative.go
package dataflowbench

func dfb_source() string { // DFB-SOURCE: nested-access-path-input
	return "tainted"
}

func dfb_sink(value string) {} // DFB-SINK: nested-access-path-sink

type Level3 struct {
	Value string
	Other string
}

type Level2 struct {
	C Level3
}

type Level1 struct {
	B Level2
}

func run() {
	a := Level1{B: Level2{C: Level3{Value: "clean", Other: "clean"}}}
	a.B.C.Value = dfb_source() // DFB-WITNESS: nested-access-path-store
	dfb_sink(a.B.C.Other)
}

File: cases/taint/rust/nested-access-path-negative/nested_access_path_negative.rs
fn dfb_source() -> i32 { // DFB-SOURCE: nested-access-path-input
    1
}

fn dfb_sink(value: i32) {} // DFB-SINK: nested-access-path-sink

struct LevelThree {
    value: i32,
    other: i32,
}

struct LevelTwo {
    c: LevelThree,
}

struct LevelOne {
    b: LevelTwo,
}

fn run() {
    let mut a = LevelOne {
        b: LevelTwo {
            c: LevelThree { value: 0, other: 0 },
        },
    };
    a.b.c.value = dfb_source(); // DFB-WITNESS: nested-access-path-store
    dfb_sink(a.b.c.other);
}
